python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -17,7 +17,25 @@
 from dagster._core.definitions.resource_definition import dagster_maintained_resource
 from dagster._core.storage.event_log.sql_event_log import SqlDbConnection
 from dagster._utils.cached_method import cached_method
-from pydantic import Field, root_validator, validator
+from pydantic import Field, validator
+
+try:
+    # Pydantic 2.x
+    from pydantic import model_validator  # type: ignore
+except ImportError:
+    # Pydantic 1.x
+    from pydantic import root_validator
+
+    def model_validator(mode="before"):
+        def _decorate(func):
+            return (
+                root_validator(pre=True)(func)
+                if mode == "before"
+                else root_validator(post=False)(func)
+            )
+
+        return _decorate
```

Comment:
this is copypaste from the databricks one right? This should be in common lib if so